### PR TITLE
show extensions when pack inspect

### LINKED ISSUE #1486: `pack inspect` when image built with extensions should show the extensions
This output should include `Extensions` if extensions were used:

```
$  pack inspect hello-extensions
Inspecting image: hello-extensions

REMOTE:
(not present)

LOCAL:

Stack: io.buildpacks.samples.stacks.alpine

Base Image:
  Reference: 17eeba339127d21e62348fe7692bc6ba04f988be690d0cecd249e940efb90141
  Top Layer: sha256:ee08c026c070425ca3aa6e49e07338ab4eee589f7ce35bb921832b6013d39f49

Run Images:
  cnbs/sample-stack-run:alpine

Buildpacks:
  ID                              VERSION        HOMEPAGE
  samples/hello-extensions        0.0.1          https://github.com/buildpacks/samples/tree/main/buildpacks/hello-extensions

Processes:
  TYPE                  SHELL        COMMAND        ARGS
  curl (default)                     curl           --version
```

### PR DESCRIPTION
**Pull request by itsdarshankumar**
## Summary
pack insepct command output should include Extensions if extensions were used while building the image

## Output
image

#### Before
image

#### After
Image With Extension:
image

Image Without Extension:
image

## Related

Resolves #1486

### CODE REVIEW

File: internal/inspectimage/bom_display.go [hunk 1]
```diff
@@ -14,20 +14,42 @@ type BOMDisplay struct {
...
+}
+
 type BOMEntryDisplay struct {
 	Name      string                 `toml:"name" json:"name" yaml:"name"`
 	Version   string                 `toml:"version,omitempty" json:"version,omitempty" yaml:"version,omitempty"`
 	Metadata  map[string]interface{} `toml:"metadata" json:"metadata" yaml:"metadata"`
 	Buildpack dist.ModuleRef         `json:"buildpacks" yaml:"buildpacks" toml:"buildpacks"`
 }
 
+type BOMEntryWithExtensionDisplay struct {
```
Comment (jjbustamante) [inline internal/inspectimage/bom_display.go:31]: I have the feeling that this should name `BOMEntryWithExtension`? I mean, a Display contains entries of this struct, right?

File: internal/inspectimage/bom_display.go [hunk 2]
```diff
@@ -48,3 +70,31 @@ func displayBOM(bom []buildpack.BOMEntry) []BOMEntryDisplay {
 
 	return result
 }
+
+func displayBOMWithExtension(bom []buildpack.BOMEntry) []BOMEntryWithExtensionDisplay {
```
Comment (jjbustamante) [inline internal/inspectimage/bom_display.go:74]: Based on my previous comment, if the struct name is changed to `BOMEntryWithExtension` then the method's name make sense to what is returning.

File: internal/inspectimage/info_display.go [hunk 3]
```diff
@@ -47,23 +47,48 @@ type InfoDisplay struct {
...
+	Local     *InfoWithExtensionDisplay `json:"local_info" yaml:"local_info" toml:"local_info"`
+}
+
+func NewInfoDisplay(info *client.ImageInfo, infoWithExtension *client.ImageWithExtensionInfo, generalInfo GeneralInfo) (*InfoDisplay, *InfoWithExtensionDisplay) {
+	if info == nil && infoWithExtension == nil {
+		return nil, nil
+	}
 	if info == nil {
-		return nil
+		return nil, &InfoWithExtensionDisplay{
```
Comment (jjbustamante) [inline internal/inspectimage/info_display.go:76]: Nice!

File: pkg/client/inspect_image.go [hunk 4]
```diff
@@ -52,6 +52,48 @@ type ImageInfo struct {
...
+	// Stack Identifier used when building this image
+	StackID string
+
+	// List of buildpacks that passed detection, ran their build
+	// phases and made a contribution to this image.
+	Buildpacks []buildpack.GroupElement
+
+	// List of extensions that passed detection, ran their generate
+	// phases and made a contribution to this image.
+	Extensions []buildpack.GroupElement
```
Comment (jjbustamante) [inline pkg/client/inspect_image.go:67]: @itsdarshankumar  I have a doubt, which problems did you face if you just add  `Extensions []buildpack.GroupElement` into `ImageInfo` struct? I mean, it an `ImageInfo` could have extension, then is this new struct totally necessary? did you face issues trying to print the output in that way? I just want to understand why do we need this new one.

**Review by jjbustamante — APPROVED** [review]
Nice work @itsdarshankumar.

Thanks a lot for your contributions!

**Review by jkutner — APPROVED** [review]

File: internal/inspectimage/bom_display.go [hunk 5]
```diff
@@ -19,12 +19,16 @@ type BOMEntryDisplay struct {
 	Version   string                 `toml:"version,omitempty" json:"version,omitempty" yaml:"version,omitempty"`
 	Metadata  map[string]interface{} `toml:"metadata" json:"metadata" yaml:"metadata"`
 	Buildpack dist.ModuleRef         `json:"buildpacks" yaml:"buildpacks" toml:"buildpacks"`
+	Extension dist.ModuleRef         `json:"extensions" yaml:"extensions" toml:"extensions"`
```
Comment (natalieparellano) [inline internal/inspectimage/bom_display.go:22]: I had the occasion to go back and look at this PR, and actually I don't think this is what we want. A "bom entry" is an outdated concept that only applies to buildpacks. It appears that we're basically just repeating information but under the header "extensions"... (cc @jjbustamante )

**Pull request merged**